nautobot_circuit_maintenance/jobs/site_search.py

```diff
@@ -0,0 +1,159 @@
+"""Site searching Job definition."""
+import collections
+from datetime import datetime, date
+from typing import NamedTuple
+
+from django.conf import settings
+
+from nautobot.extras.jobs import Job
+from nautobot.circuits.models import Circuit
+
+from nautobot_circuit_maintenance.models import CircuitMaintenance
+
+PLUGIN_SETTINGS = settings.PLUGINS_CONFIG.get("nautobot_circuit_maintenance", {})
+CIRCUIT_MAINTENANCE_TAG_COLOR = "Purple"
+
+name = "Circuit Maintenance"  # pylint: disable=invalid-name
+
+
+class DateTimeRange(NamedTuple):
+    """Range class for use within the check for overlap.
+
+    Args:
+        NamedTuple (NamedTupe): Parent class
+    """
+
+    start: datetime
+    end: datetime
+
+
+def check_for_overlap(record1: CircuitMaintenance, record2: CircuitMaintenance):
```

Comment:
thinking alound, would make sense to generalize this function to `**records`, so you could find overlapping for more than only 2 maintenances? It works for the current case, and it will work for future cases where more than 2 maintenances could be taken into account, in one shot.
It's simply adding a for loop for ranges, and get the same max and min, no?

Reply:
Maybe. In the longer run I think we need to compare the number of overlapping records to the number of circuits. If there are 4 circuits to a site, then as long as there are no more than 3 maintenances that overlap, we are good. May need some logic thinking on that one.

Reply:
ok, let's leave for another iteration